lib/datacite/validators/attributes_validator.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'json_schemer'
+require 'pathname'
+
+module Datacite
+  module Validators
+    # Perform validation against openapi definition for the DataCite API
+    class AttributesValidator
+      def self.validate(attributes)
+        json_attributes = JSON.parse(attributes.to_json)
+        schemer = JSONSchemer.schema(json_schema_path)
+        results = schemer.validate(json_attributes)
+        return true if results.first.nil?
+
+        raise ValidationError, results.map { |result| result.fetch('error') }.join('\n')
```

Comment:
What is `results` here? Is it a plain ol' array or an instance of a class from the JSON schemer gem returned by the validation or?

I'm wondering why we might be checking the value of `results.first` here.

Reply:
results is always `Enumerator`. I thought I got `.first.nil?` from the README but I'm not seeing it, so it may have been me working through it. I'll see if something like `.empty?` is more appropriate.